cmd/resource-topology-exporter/main.go

```diff
@@ -41,176 +51,103 @@ func main() {
 
-	// Parse argument values as usable types.
-	nrtupdaterArgs.NoPublish = arguments["--no-publish"].(bool)
-	nrtupdaterArgs.Oneshot = arguments["--oneshot"].(bool)
-	if ns, ok := arguments["--export-namespace"].(string); ok {
-		nrtupdaterArgs.Namespace = ns
-	}
-	if hostname, ok := arguments["--hostname"].(string); ok {
-		nrtupdaterArgs.Hostname = hostname
+	flags.StringVar(&configPath, "config", "/etc/resource-topology-exporter/config.yaml", "Configuration file path. Use this to set the exclude list.")
+	conf, err := readConfig(configPath)
+	if err != nil {
+		return pArgs, fmt.Errorf("error getting exclude list from the configuration: %v", err)
 	}
-	if nrtupdaterArgs.Hostname == "" {
-		var err error
-		nrtupdaterArgs.Hostname = os.Getenv("NODE_NAME")
-		if nrtupdaterArgs.Hostname == "" {
-			nrtupdaterArgs.Hostname, err = os.Hostname()
-			if err != nil {
-				return nrtupdaterArgs, resourcemonitorArgs, rteArgs, fmt.Errorf("error getting the host name: %w", err)
-			}
-		}
+	if len(conf.ExcludeList) != 0 {
+		pArgs.resourcemonitor.ExcludeList.ExcludeList = conf.ExcludeList
+		log.Printf("using exclude list:\n%s", pArgs.resourcemonitor.ExcludeList.String())
 	}
 
-	resourcemonitorArgs.SleepInterval, err = time.ParseDuration(arguments["--sleep-interval"].(string))
+	flags.BoolVar(&pArgs.rte.Debug, "debug", false, " Enable debug output.")
+	flags.StringVar(&pArgs.rte.TopologyManagerPolicy, "topology-manager-policy", defaultTmPolicy(), "Explicitly set the topology manager policy instead of reading from the kubelet.")
+
+	pArgs.rte.ReferenceContainer = &podrescli.ContainerIdent{}
+	flags.Var(pArgs.rte.ReferenceContainer, "reference-container", "Reference container, used to learn about the shared cpu pool\n See: https://github.com/kubernetes/kubernetes/issues/102190\n format of spec is namespace/podname/containername.\n Alternatively, you can use the env vars REFERENCE_NAMESPACE, REFERENCE_POD_NAME, REFERENCE_CONTAINER_NAME.")
+
+	pArgs.version = flags.Bool("version", false, "Output version and exit")
+
+	err = flags.Parse(args)
 	if err != nil {
-		return nrtupdaterArgs, resourcemonitorArgs, rteArgs, fmt.Errorf("invalid --sleep-interval specified: %w", err)
-	}
-	if ns, ok := arguments["--watch-namespace"].(string); ok {
-		resourcemonitorArgs.Namespace = ns
-	}
-	if kubeletConfigPath, ok := arguments["--kubelet-config-file"].(string); ok {
-		resourcemonitorArgs.KubeletConfigFile = kubeletConfigPath
-	}
-	resourcemonitorArgs.SysfsRoot = arguments["--sysfs"].(string)
-	if path, ok := arguments["--podresources-socket"].(string); ok {
-		resourcemonitorArgs.PodResourceSocketPath = path
+		return pArgs, err
 	}
 
-	if kubeletStateDirs, ok := arguments["--kubelet-state-dir"].([]string); ok {
-		resourcemonitorArgs.KubeletStateDirs = kubeletStateDirs
+	if reflect.DeepEqual(*pArgs.rte.ReferenceContainer, podrescli.ContainerIdent{}) {
```

Comment:
just check if the contents are non-zero value and/or add a helper method into `podrescli.ContainerIdent`. maybe `IsEmpty`.

Reply:
This will allow us to expand ```ReferenceContainer``` without worries about maintaining the ```IsEmpty``` or adding additional comparisons don't you think?

Reply:
I strongly prefer an explicit method (one-liner anyway) unless there is another reason I don't see atm. The method itself can use `reflect.DeepEqual` internally if needed (not a fan of it but I'm open to try)